File: src/main/node/net/ecma/Library/Tests/Components/Navigation/ClonedLOD.mjs
'use strict';
var X3D = require('./x3d.mjs');
try {
	var fs = require('fs');
} catch (e) {
console.log("Problems loading fs. On browser?",e);
}
var SFString = require('./x3d.mjs');
var SFNode = require('./x3d.mjs');
var head = require('./x3d.mjs');
var MFNode = require('./x3d.mjs');
var meta = require('./x3d.mjs');
var Scene = require('./x3d.mjs');
var Transform = require('./x3d.mjs');
var SFVec3f = require('./x3d.mjs');
var LOD = require('./x3d.mjs');
var MFFloat = require('./x3d.mjs');
var Shape = require('./x3d.mjs');
var Appearance = require('./x3d.mjs');
var Material = require('./x3d.mjs');
var SFFloat = require('./x3d.mjs');
var SFColor = require('./x3d.mjs');
var Box = require('./x3d.mjs');
var Viewpoint = require('./x3d.mjs');
var X3D0 =  new X3D({

      profile : new SFString("Full"),
      version : new SFString("3.3"),
      head : new SFNode(
        new head({
          meta : new MFNode([
            new meta({
              name : new SFString("comment"),
              content : new SFString("World of Titania")}),

            new meta({
              name : new SFString("created"),
              content : new SFString("Thu, 10 Sep 2015 07:10:33 GMT")}),

            new meta({
              name : new SFString("creator"),
              content : new SFString("Holger Seelig")}),

            new meta({
              name : new SFString("generator"),
              content : new SFString("Titania V1.1.0, http://titania.create3000.de")}),

            new meta({
              name : new SFString("modified"),
              content : new SFString("Thu, 10 Sep 2015 07:10:33 GMT")})])})),
      Scene : new SFNode(
        new Scene({
          children : new MFNode([
            new Transform({
              DEF : new SFString("_1"),
              translation : new SFVec3f([-3.07774,0,0]),
              children : new MFNode([
                new LOD({
                  DEF : new SFString("_2"),
                  range : new MFFloat([20]),
                  children : new MFNode([
                    new Transform({
                      DEF : new SFString("Box_1"),
                      children : new MFNode([
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  DEF : new SFString("Rococo12_1"),
                                  ambientIntensity : new SFFloat(0.226102),
                                  diffuseColor : new SFColor([0.904409,0.146366,0.208768]),
                                  specularColor : new SFColor([0.0955906,0.0955906,0.0955906]),
                                  shininess : new SFFloat(0.078125)}))})),
                          geometry : new SFNode(
                            new Box({}))})])}),

                    new Transform({
                      DEF : new SFString("Box_2"),
                      children : new MFNode([
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  DEF : new SFString("Rococo28_1"),
                                  ambientIntensity : new SFFloat(0.226102),
                                  diffuseColor : new SFColor([0.279645,0.904409,0.511589]),
                                  specularColor : new SFColor([0.0955906,0.0955906,0.0955906]),
                                  shininess : new SFFloat(0.078125)}))})),
                          geometry : new SFNode(
                            new Box({}))})])})])})])}),

            new Transform({
              translation : new SFVec3f([3.35422,0,-16.1449]),
              children : new MFNode([
                new LOD({
                  USE : new SFString("_2")})])}),

            new Viewpoint({
              DEF : new SFString("_3"),
              description : new SFString("Home")})])}))});
console.log(X3D0.toXMLNode());

File: src/main/c/net/coderextreme/Library/Tests/Components/Navigation/ClonedLOD.c
#include "C:/x3d-code/www.web3d.org/x3d/languages/c/Concretes.h"
void main(int argc, char ** argv) {
Browser browser = X3D.getBrowser();
X3D X3D0;
X3D0.profile = "Full";
X3D0.version = "3.3";
head head1 = createNode("head");
meta meta2 = createNode("meta");
meta2.name = "comment";
meta2.content = "World of Titania";
head1.meta = new MFNode();

head1.meta[0] = meta2;

meta meta3 = createNode("meta");
meta3.name = "created";
meta3.content = "Thu, 10 Sep 2015 07:10:33 GMT";
head1.meta[1] = meta3;

meta meta4 = createNode("meta");
meta4.name = "creator";
meta4.content = "Holger Seelig";
head1.meta[2] = meta4;

meta meta5 = createNode("meta");
meta5.name = "generator";
meta5.content = "Titania V1.1.0, http://titania.create3000.de";
head1.meta[3] = meta5;

meta meta6 = createNode("meta");
meta6.name = "modified";
meta6.content = "Thu, 10 Sep 2015 07:10:33 GMT";
head1.meta[4] = meta6;

head = head1;

Transform Transform8 = createNode("Transform");
Transform8.DEF = "_1";
Transform8.translation = new SFVec3f(new float[-3.07774,0,0]);
LOD LOD9 = createNode("LOD");
LOD9.DEF = "_2";
LOD9.range = new MFFloat(new float[20]);
Transform Transform10 = createNode("Transform");
Transform10.DEF = "Box_1";
Shape Shape11 = createNode("Shape");
Appearance Appearance12 = createNode("Appearance");
Material Material13 = createNode("Material");
Material13.DEF = "Rococo12_1";
Material13.ambientIntensity = 0.226102;
Material13.diffuseColor = new SFColor(new float[0.904409,0.146366,0.208768]);
Material13.specularColor = new SFColor(new float[0.0955906,0.0955906,0.0955906]);
Material13.shininess = 0.078125;
Appearance12.material = Material13;

Shape11.appearance = Appearance12;

Box Box14 = createNode("Box");
Shape11.geometry = Box14;

Transform10.child = new undefined();

Transform10.child[0] = Shape11;

LOD9.children = new MFNode();

LOD9.children[0] = Transform10;

Transform Transform15 = createNode("Transform");
Transform15.DEF = "Box_2";
Shape Shape16 = createNode("Shape");
Appearance Appearance17 = createNode("Appearance");
Material Material18 = createNode("Material");
Material18.DEF = "Rococo28_1";
Material18.ambientIntensity = 0.226102;
Material18.diffuseColor = new SFColor(new float[0.279645,0.904409,0.511589]);
Material18.specularColor = new SFColor(new float[0.0955906,0.0955906,0.0955906]);
Material18.shininess = 0.078125;
Appearance17.material = Material18;

Shape16.appearance = Appearance17;

Box Box19 = createNode("Box");
Shape16.geometry = Box19;

Transform15.child = new undefined();

Transform15.child[0] = Shape16;

LOD9.children[1] = Transform15;

Transform8.children = new MFNode();

Transform8.children[0] = LOD9;

children = new MFNode();

children[0] = Transform8;

Transform Transform20 = createNode("Transform");
Transform20.translation = new SFVec3f(new float[3.35422,0,-16.1449]);
LOD LOD21 = createNode("LOD");
LOD21.USE = "_2";
Transform20.children = new MFNode();

Transform20.children[0] = LOD21;

children[1] = Transform20;

Viewpoint Viewpoint22 = createNode("Viewpoint");
Viewpoint22.DEF = "_3";
Viewpoint22.description = "Home";
children[2] = Viewpoint22;

}
